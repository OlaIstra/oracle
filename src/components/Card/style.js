import styled from "styled-components";
import { color, radius, padding, fontSize, height } from "../../styles/index";

export const CardStyle = styled.div`
  position: relative;
  margin: 0 auto 50px;
  display: flex;
  flex-direction: column;
  width: 100%;
  background-color: ${color.white};
  border-radius: ${radius.small}px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 3px 1px -2px rgba(0, 0, 0, 0.12), 0 1px 5px 0 rgba(0, 0, 0, 0.2);
`;

export const CardImageStyle = styled.div`
  position: relative;
  display: flex;
  background: ${color.blue};
  flex-direction: column;
  align-items: center;
  padding: ${padding.small}px ${padding.normal}px;
  margin-bottom: 20px;
`;

export const CardImgStyle = styled.img`
  display: block;
  border-radius: 50%;
  position: relative;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  width: 120px;
  height: 120px;
`;

export const CardContentStyle = styled.div`
  margin-bottom: 20px;
  padding: ${padding.medium};
  border-radius: 0 0 ${radius.small}px ${radius.small}px;
`;

export const CardNameStyle = styled.div`
  bottom: 10px;
  left: 10px;
  max-width: 100%;
  padding: ${padding.small}px;
  color: ${color.white};
  font-size: ${fontSize.medium}px;
`;

export const CardActiveStyle = styled.div`
  height: ${height.medium}px;
  display: flex;
  justify-content: space-around;
`;
