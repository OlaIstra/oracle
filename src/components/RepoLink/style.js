import styled from "styled-components";
import { device, padding, color, radius } from "../../styles/index";

export const ShadowDivStyle = styled.div`
  display: flex;
  justify-content: space-between;
  width: auto;
  padding: ${padding.medium}px;
  margin: 10px auto;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: ${color.white};
  border-radius: ${radius.normal}px;
  box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
    0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);

  &:hover {
    background: ${color.blueDark};
    color: ${color.white};
  }
  &.searchRepos {
    background: ${color.greyLight};
    color: inherit;
  }
  @media ${device.mobileL} {
    flex-direction: column;
  }
`;

export const RepoLinkStyle = styled.a`
  cursor: pointer;
  text-decoration: none;
  color: inherit;
`;
