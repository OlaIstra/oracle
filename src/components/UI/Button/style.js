import styled from "styled-components";
import {
  device,
  padding,
  height,
  lineHeight,
  radius,
  fontSize,
  color
} from "../../../styles/index";

export const ButtonStyle = styled.button`
  height: ${height.small}px;
  line-height: ${lineHeight.large}px;
  padding: 0 ${padding.normal}px;
  text-transform: uppercase;
  border: none;
  border-radius: ${radius.small}px;
  font-size: ${fontSize.small}px;
  outline: 0;
  text-decoration: none;
  color: ${color.white};
  text-align: center;
  letter-spacing: 0.5px;
  cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
  background: ${props => (props.disabled ? color.greyLight : color.blue)};
  pointer-events: ${props => (props.disabled ? "none" : "all")};
  @media ${device.mobileL} {
    margin-bottom: 20px;
  }
`;
