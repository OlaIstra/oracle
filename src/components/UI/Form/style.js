import styled from "styled-components";
import { device } from "../../../styles/index";

export const FormStyle = styled.form`
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 2rem auto;
  height: 100px;
  width: 100%;
  border-radius: 4px;
  box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
    0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);
  @media ${device.mobileL} {
    flex-direction: column;
    height: auto;
  }
`;
