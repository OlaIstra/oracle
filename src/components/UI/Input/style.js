import styled from "styled-components";
import { color, device } from "../../../styles/index";

export const InputWrapper = styled.div`
  position: relative;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  padding: 0 0.75rem;
  min-height: 1px;
  width: 50%;
  @media ${device.mobileL} {
    width: 90%;
  }
`;

export const InputStyle = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 1px solid ${color.greyLight};
  border-radius: 0;
  outline: none;
  height: 50px;
  width: 100%;
  font-size: 20px;
  margin: 0 0 8px 0;
  padding: 0;
  box-shadow: none;
  box-sizing: content-box;
`;
