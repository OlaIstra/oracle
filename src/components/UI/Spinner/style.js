import styled, { keyframes } from "styled-components";
import { color } from "../../../styles/index";

const rotate = keyframes`
  0%,
  80%,
  100% {
    box-shadow: 0 0;
    height: 4em;
  }
  40% {
    box-shadow: 0 -2em;
    height: 5em;
  }
`;

export const SpinnerStyle = styled.div`
  color: ${color.orange};
  text-indent: -9999em;
  margin: 88px auto;
  position: relative;
  font-size: 11px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
  background: ${color.orange};
  -webkit-animation: ${rotate} 1s infinite ease-in-out;
  animation: ${rotate} 1s infinite ease-in-out;
  width: 1em;
  height: 4em;
  :before {
    background: ${color.orange};
    -webkit-animation: ${rotate} 1s infinite ease-in-out;
    animation: ${rotate} 1s infinite ease-in-out;
    width: 1em;
    height: 4em;
    position: absolute;
    top: 0;
    content: "";
    left: -1.5em;
    -webkit-animation-delay: -0.32s;
    animation-delay: -0.32s;
  }
  :after {
    background: ${color.orange};
    -webkit-animation: ${rotate} 1s infinite ease-in-out;
    animation: ${rotate} 1s infinite ease-in-out;
    width: 1em;
    height: 4em;
    position: absolute;
    top: 0;
    content: "";
    left: 1.5em;
  }
`;
