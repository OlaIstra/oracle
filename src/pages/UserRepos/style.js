import styled from "styled-components";
import { device } from "../../styles/index";

export const UserReposStyle = styled.div`
  position: relative;
  width: 100%;
  margin: 0 auto;
`;

export const SortByRateStyle = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  label {
    margin-right: 10px;
  }
  input {
    margin-bottom: 0px;
  }
  @media ${device.mobileL} {
    justify-content: center;
  }
`;
