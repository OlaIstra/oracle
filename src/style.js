import styled from "styled-components";
import { device } from "./styles/index";

export const AppStyle = styled.div`
  max-width: 400px;
  margin: 0 auto;
  text-align: center;
  @media ${device.mobileL} {
    max-width: 90%;
  }
`;
