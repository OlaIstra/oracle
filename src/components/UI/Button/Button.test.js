import React from "react";
import { ButtonStyle } from "./style";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { color } from "../../../styles/index";

describe("ButtonStyle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ButtonStyle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("disabled if props.disabled", () => {
    const tree = renderer.create(<ButtonStyle disabled />).toJSON();
    expect(tree).toHaveStyleRule("cursor", "not-allowed");
    expect(tree).toHaveStyleRule("background", color.greyBlack);
    expect(tree).toHaveStyleRule("pointer-events", "none");
  });
});
