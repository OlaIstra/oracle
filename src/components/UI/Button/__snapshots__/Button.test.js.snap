// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyle renders correctly 1`] = `
.c0 {
  height: 40px;
  line-height: 35px;
  padding: 0 15px;
  text-transform: uppercase;
  border: none;
  border-radius: 2px;
  font-size: 14px;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  cursor: pointer;
  background: #262244;
  pointer-events: all;
}

@media (max-width:425px) {
  .c0 {
    margin-bottom: 20px;
  }
}

<button
  className="c0"
/>
`;
